.pipelines/templates/e2e-test-kind.yaml

```diff
@@ -1,50 +1,43 @@
 jobs:
   - job:
-    displayName: e2e_test_kind/helm
     timeoutInMinutes: 10
     cancelTimeoutInMinutes: 5
+    dependsOn:
+    - lint
+    - scan_images
+    - unit_test
+    - build
     workspace:
       clean: all
     variables:
     - group: csi-secrets-store-e2e-kind
-    steps:
-      - script: |
-          export REGISTRY="e2e"
-          export IMAGE_VERSION=e2e-$(git rev-parse --short HEAD)
-          echo "Image version: ${IMAGE_VERSION}"
-          echo "##vso[task.setvariable variable=IMAGE_VERSION]${IMAGE_VERSION}"
-          echo "##vso[task.setvariable variable=REGISTRY]${REGISTRY}"
-          make e2e-bootstrap
-        displayName: "Build image"
-        env:
-          CI_KIND_CLUSTER: true
-      - script: |
-          make e2e-test
-        displayName: "Run e2e tests on kind cluster"
-        env:
-          AZURE_CLIENT_ID: $(AZURE_CLIENT_ID)
-          AZURE_CLIENT_SECRET: $(AZURE_CLIENT_SECRET)
-          KEY_NAME: $(KEY_NAME)
-          KEY_VERSION: $(KEY_VERSION)
-          KEYVAULT_NAME: $(KEYVAULT_NAME)
-          SECRET_NAME: $(SECRET_NAME)
-          TENANT_ID: $(TENANT_ID)
-          CI_KIND_CLUSTER: true
-          AZURE_ENVIRONMENT_FILEPATH: "/etc/kubernetes/custom_environment.json"
+    strategy:
+      matrix:
```

Comment:
Why we are running with 4 k8s version and not 3?

Reply:
1.19 is still not EOL and 1.22 is the latest. Is there a reason we only want to run 3?

Reply:
We generally support n-2 right? (total 3 releases)

Reply:
1.19, 1.20 and 1.21 is the current supported version. 1.22 is the latest which we're also running test against here. I think it's good to catch regressions with latest kubernetes versions earlier when they're available. WDYT?

Reply:
Ok. Sounds good.

Also, Should we document explicit k8s version we are supporting? Something like _currently supported k8s versions_. What say?

Reply:
sgtm! We can add the current supported version and versions we test in doc in our docs.